programs/ziti-edge-tunnel/include/log_utils.c

```diff
@@ -0,0 +1,315 @@
+    );
+
+    if ( ziti_tunneler_log != NULL) {
+        fputc('\n', ziti_tunneler_log);
+        fprintf(ziti_tunneler_log, "[%s] %7s %s ", curr_time, parse_level(level), loc);
+        fwrite(msg, 1, msglen, ziti_tunneler_log);
+    }
+
+    if(multi_writer) {
+        printf("\n[%s] %7s %s %.*s", curr_time, parse_level(level), loc, msglen, msg);
+    }
+
+}
+
+bool open_log(char* log_filename) {
+    if((ziti_tunneler_log=fopen(log_filename,"a")) == NULL) {
+        printf("Could not open logs file %s", log_filename);
+        return false;
+    }
+    return true;
+}
+
+void close_log() {
+    if (ziti_tunneler_log != NULL) {
+        fclose(ziti_tunneler_log);
+        ziti_tunneler_log = NULL;
+    }
+    if (log_filename != NULL) {
+        free(log_filename);
+        log_filename = NULL;
+    }
+}
+
+void stop_log_check() {
+    uv_check_stop(log_flusher);
+}
+
+void rotate_log() {
+    close_log();
+
+    uv_timeval64_t file_time;
+    uv_gettimeofday(&file_time);
+    struct tm* orig_time = gmtime(&file_time.tv_sec);
+    memcpy(start_time, orig_time, sizeof(struct tm));
+    log_filename = get_log_filename();
+
+    open_log(log_filename);
+}
+
+void delete_older_logs(uv_loop_t *ziti_loop) {
+    char curr_path[FILENAME_MAX]; //create string buffer to hold path
+#if _WIN32
+    GetCurrentDir( curr_path, FILENAME_MAX );
+#else
+    sprintf(curr_path, "%s", "/tmp");
+#endif
+
+    char log_path[FILENAME_MAX];
+    sprintf(log_path, "%s/logs", curr_path);
+
+    uv_fs_t fs;
+    int rc = uv_fs_scandir(ziti_loop, &fs, log_path, 0, NULL);
+    if (rc < 7) {
+        if (rc < 0) {
+            ZITI_LOG(ERROR, "failed to scan dir[%s]: %d/%s", log_path, rc, uv_strerror(rc));
+        } else {
+            ZITI_LOG(TRACE, "Log files count in [%s] is %d, not deleting log files.", log_path, rc);
+            uv_fs_req_cleanup(&fs);
+        }
+        return;
+    }
+
+    char **log_files = calloc(rc + 1 , sizeof(char *));
+    uv_dirent_t file;
+    int rotation_cnt = 0;
+    while (uv_fs_scandir_next(&fs, &file) == 0) {
+        ZITI_LOG(TRACE, "log file = %s %d", file.name, file.type);
+
+        if (file.type == UV_DIRENT_FILE) {
+            if (memcmp(file.name, log_filename_base, sizeof(log_filename_base)) == 0) {
+                log_files[rotation_cnt] = strdup(file.name);
+                rotation_cnt++;
+            }
+        }
+    }
+
+    char currpath[FILENAME_MAX]; //create string buffer to hold path
+#if _WIN32
+    GetCurrentDir( currpath, FILENAME_MAX );
+#else
+    sprintf(curr_path, "%s", "/tmp");
+#endif
+
+    char logpath[FILENAME_MAX];
+    sprintf(logpath, "%s/logs", currpath);
+    int rotation_index = rotation_cnt;
+    while (rotation_index > rotation_count) {
+        char *old_log= calloc(FILENAME_MAX, sizeof(char));
+        int old_idx = -1;
+
+        for(int idx =0; idx < rotation_cnt; idx++) {
+            if (*old_log == NULL && log_files[idx]) {
```

Comment:
`*old_log` is a character, not a pointer. comparing it to NULL is probably not what you want. Notice that CLion flags this as a warning. The remaining tests for `*old_log` have the same issue. Do you intend to compare with `\0` (end of string)?